Fix Java 2 security test failures

jhanders34 opened the pull request:
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

jhanders34 committed:
Fix Java 2 security test failures

jhanders34 commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=f9aae49f-39a7-4d66-8c17-43b079957ecc) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

tjwatson reviewed (dismissed):
I approve the `javaPermission` configuration for the `derby/` folder that was added for the path test.

I'm not sure how the `null` checks relate in the `JDBCDerbyServlet` class.  An NPE probably got exposed when the security check failed for derby code?  Regardless, the `null` check looks fine to me.

Azquelt reviewed (dismissed):
Thanks for picking this up.

Azquelt commented on dev/com.ibm.ws.jdbc_fat_derby/test-applications/jdbcapp/src/jdbc/fat/derby/web/JDBCDerbyServlet.java:
```diff
@@ -207,9 +212,11 @@ public void testTNTransationEnlistment() throws Throwable {
             if (!"Olmsted".equals(value))
                 throw new Exception("Incorrect value: " + value);
         } finally {
-            result.close();
-            dropTable(con, CITYTABLE);
-            con.close();
+            if (con != null) {
+                result.close();
+                dropTable(con, CITYTABLE);
+                con = null;
+	    }
```
Is the change from `con.close()` to `con = null` deliberate here?

jhanders34 replied:
Yes that is wrong.  I fixed it.

jhanders34 commented:
> I'm not sure how the `null` checks relate in the `JDBCDerbyServlet` class. An NPE probably got exposed when the security check failed for derby code? Regardless, the `null` check looks fine to me.

Yes, I updated to do the `null` checks because it hid the actual exception that caused `con` variables not to get set so in the finally an NPE happens and the real exception was lost which was the AccessControlException.  To make it easier to find the problem in the future, I updated the Servlet to no longer get NPEs if getting the connection fails.

jhanders34 commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=8bf53b6d-6b22-42be-a09f-c1cdefe34847) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._
